includes/sanitizers/class-amp-blacklist-sanitizer.php

```diff
@@ -41,18 +48,28 @@ private function strip_attributes_recursive( $node, $bad_attributes, $bad_protoc
 				if ( 0 === stripos( $attribute_name, 'on' ) && $attribute_name != 'on' ) {
 					$node->removeAttribute( $attribute_name );
 					continue;
-				} elseif ( 'href' === $attribute_name ) {
+				} elseif ( 'href' === $attribute_name && 'a' !== $node_name ) {
 					$protocol = strtok( $attribute->value, ':' );
 					if ( in_array( $protocol, $bad_protocols ) ) {
 						$node->removeAttribute( $attribute_name );
 						continue;
 					}
 				} elseif ( 'a' === $node_name ) {
-					$this->sanitize_a_attribute( $node, $attribute );
+					// Sanitize the tag, but remove it entirely if the href is invalid.
+					// Children will be preserved as part of the parent.
+					if ( false === $this->sanitize_a_attribute( $node, $attribute ) ) {
```

Comment:
Fair point indeed. I've handled this and also added the scenario of `<a name="section2"></a>` both to the validation and unit tests.